### PR TITLE
[mbedtls] remove GSDK mbedtls config template from includes

### PR DESCRIPTION
**Pull request by lmnotran**
This fixes a build failure that occurs when `-DOT_COAPS=ON`. 

Including this directory was causing the 'mbedtls_config.h' from that directory to be loaded instead of the one
in `src/src`

This PR also adds a copy of `psa_crypto_config.h` to `src/src`

### CODE REVIEW

**Comment by lmnotran** [comment]
@jwhui - Unrelated to this PR, could you set the default merge strategy in this repo to use squashed merges?

**Review by hemanth-silabs — APPROVED** [review]

**Comment by hemanth-silabs** [comment]
LGTM

**Review by romacdon — COMMENTED** [review]
LGTM

File: src/src/psa_crypto_config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#ifndef PSA_CRYPTO_CONFIG_H
```
[the remark this answers is not in the document]
Comment (lmnotran) [reply src/src/psa_crypto_config.h:1]: This is correct

**Review by romacdon — APPROVED** [review]
LGTM

**Comment by romacdon** [comment]
@jwhui can you merge this PR please?

**Review by jwhui — APPROVED** [review]

**Pull request merged**